### PR TITLE
[KOGITO-2317] - Unify Kogito Quarkus and Spring Images

### PR DESCRIPTION
**Pull request by Kaitou786**
See: https://issues.redhat.com/browse/KOGITO-2317

Signed-off-by: Tarun Khandelwal <tarkhand@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [x] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

File: controllers/build/image_stream_kogito.go [hunk 1]
```diff
@@ -37,59 +37,39 @@ const (
...
-			"openshift.io/display-name":          "Platform for building Kogito based on Quarkus",
-		},
-		infrastructure.KogitoSpringBootUbi8Image: {
-			"openshift.io/provider-display-name": "KIE Group",
-			"openshift.io/display-name":          "Runtime image for Kogito based on SpringBoot",
-		},
-		infrastructure.KogitoSpringBootUbi8s2iImage: {
-			"openshift.io/provider-display-name": "KIE Group",
-			"openshift.io/display-name":          "Platform for building Kogito based on SpringBoot",
+			"openshift.io/display-name":          "Platform for building Kogito based on Quarkus or Spring Boot",
```
Comment (sutaakar) [inline controllers/build/image_stream_kogito.go:50]: Nitpick (also in other places):
```suggestion
			"openshift.io/display-name":          "Platform for building Kogito based on Quarkus or SpringBoot",
```

File: test/framework/kogitobuild.go [hunk 2]
```diff
@@ -139,7 +141,6 @@ func getKogitoBuildImage(imageName string) string {
 	image := v1beta1.Image{
 		Domain:    config.GetBuildImageRegistry(),
 		Namespace: config.GetBuildImageNamespace(),
-		Tag:       config.GetBuildImageVersion(),
```
Comment (sutaakar) [inline test/framework/kogitobuild.go:142]: Why is this line deleted?

File: controllers/build/image_stream_kogito.go (hunk 1 above)
Comment (sutaakar) [reply controllers/build/image_stream_kogito.go:50]: or maybe not...
I always thought it is written as one word

File: test/framework/kogitobuild.go (hunk 2 above)
Comment (Kaitou786) [reply test/framework/kogitobuild.go]: mistake, reverted it back

File: controllers/build/image_stream_kogito.go (hunk 1 above)
Comment (Kaitou786) [reply controllers/build/image_stream_kogito.go]: yeah on the images I was asked to keep them separate

Comment (ricardozanini) [reply controllers/build/image_stream_kogito.go:50]: That's right: https://spring.io/projects/spring-boot

**Review by ricardozanini — APPROVED** [review]
LGTM
Codecov is really pissing you :D

**Review by sutaakar — APPROVED** [review]

**Review by radtriste — APPROVED** [review]
lgtm. one minor comment

**Review by spolti — APPROVED** [review]
